### DISCUSSION TITLE
next-auth and mongodb adapter compatibility issues?

### DISCUSSION THREAD

**Discussion by lautarosegura** [nextauthjs/next-auth#9468] [Help]
Hey! I'm trying to setup next-auth with mongodb and I'm using the official mongodb adapter. However, when I setup the adapter in my authOptions object, this error pops up:

```
 Types of property 'createUser' are incompatible.
    Type '((user: AdapterUser) => Awaitable<AdapterUser>) | undefined' is not assignable to type '((user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>) | undefined'.
      Type '(user: AdapterUser) => Awaitable<AdapterUser>' is not assignable to type '(user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>'.
        Types of parameters 'user' and 'user' are incompatible.
          Property 'id' is missing in type 'Omit<AdapterUser, "id">' but required in type 'AdapterUser'.ts(2322)
```
          
I'm currently using next-auth version 4.24.5 and @auth/mongodb-adapter version 2.0.10
Also using mongodb package with version 6.3.0

Any help will be appreciated, I've been investigating but I was not able to find a fix for this.

**Comment by campbellgoe** [comment]
If you show some code you're more likely to get a response.
import { MongoDBAdapter } from "@next-auth/mongodb-adapter"

**Comment by starthecode** [comment]
Any update on this issue, getting same issue when using - 
**import { MongoDBAdapter } from '@auth/mongodb-adapter';**
but this error get vanish when we use @next-auth/mongodb-adapter but if we use this one, we need to add --legacy-peer-deps to all other dependant.

**Comment by tchars** [comment]
```
import { MongoDBAdapter } from "@auth/mongodb-adapter";
import { Adapter } from "next-auth/adapters";
...
...
export const authOptions: NextAuthOptions = {
  adapter: MongoDBAdapter(clientPromise) as Adapter,
...
```

"**as Adapter**" worked for me
